vcd/resource_vcd_nsxt_edgegateway_bgp_configuration_test.go

```diff
@@ -0,0 +1,628 @@
+		Steps: []resource.TestStep{
+			{
+				Config: configText1,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestMatchResourceAttr("vcd_nsxt_edgegateway_bgp_configuration.testing", "id", regexp.MustCompile(`^urn:vcloud:gateway:.*$`)),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway_bgp_configuration.testing", "enabled", "false"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway_bgp_configuration.testing", "local_as_number", "65420"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway_bgp_configuration.testing", "graceful_restart_mode", "GRACEFUL_AND_HELPER"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway_bgp_configuration.testing", "graceful_restart_timer", "190"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway_bgp_configuration.testing", "stale_route_timer", "500"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway_bgp_configuration.testing", "ecmp_enabled", "false"),
+				),
+			},
+			{
+				Config: configText2,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestMatchResourceAttr("vcd_nsxt_edgegateway_bgp_configuration.testing", "id", regexp.MustCompile(`^urn:vcloud:gateway:.*$`)),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway_bgp_configuration.testing", "enabled", "true"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway_bgp_configuration.testing", "local_as_number", "65420"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway_bgp_configuration.testing", "graceful_restart_mode", "GRACEFUL_AND_HELPER"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway_bgp_configuration.testing", "graceful_restart_timer", "190"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway_bgp_configuration.testing", "stale_route_timer", "500"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway_bgp_configuration.testing", "ecmp_enabled", "true"),
+				),
+			},
+		},
+	})
+}
+
+const testAccVcdNsxtBgpConfigVdcGroupPrereqs = `
+data "vcd_vdc_group" "g1" {
+  org  = "{{.Org}}"
+  name = "{{.NsxtVdcGroup}}"
+}
+
+data "vcd_nsxt_edgegateway" "testing" {
+  org      = "{{.Org}}"
+  owner_id = data.vcd_vdc_group.g1.id
+
+  name = "{{.NsxtEdgeGwInVdcGroup}}"
+}
+`
+
+const testAccVcdNsxtBgpVdcGroupConfig1 = testAccVcdNsxtBgpConfigVdcGroupPrereqs + `
+resource "vcd_nsxt_edgegateway_bgp_configuration" "testing" {
+  org = "{{.Org}}"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.testing.id
+
+  enabled                = false
+  local_as_number        = "65420"
+  graceful_restart_mode  = "GRACEFUL_AND_HELPER"
+  graceful_restart_timer = 190
+  stale_route_timer      = 500
+  ecmp_enabled           = false
+}
+`
+
+const testAccVcdNsxtBgpVdcGroupConfig2 = testAccVcdNsxtBgpConfigVdcGroupPrereqs + `
+resource "vcd_nsxt_edgegateway_bgp_configuration" "testing" {
+  org = "{{.Org}}"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.testing.id
+
+  enabled                = true
```

Comment:
Should we add a last step to revert to disabled BGP, or was it enabled already before the test?

Reply:
I'll answer myself: terraform destroy will do it I guess :)

Reply:
Yes - Terraform destroy does it :)